controller/api/destination/endpoint_translator.go

```diff
@@ -127,11 +130,24 @@ func (et *endpointTranslator) sendFilteredUpdate(set watcher.AddressSet) {
 // topology zone. The client will only receive endpoints with the same
 // consumption zone as the node. An endpoints consumption zone is set
 // by its Hints field and can be different than its actual Topology zone.
-// when service.pec.internalTrafficPolicy is set to local, Topology Aware
+// when service.spec.internalTrafficPolicy is set to local, Topology Aware
 // Hints are not used.
 func (et *endpointTranslator) filterAddresses() watcher.AddressSet {
 	filtered := make(map[watcher.ID]watcher.Address)
-	// If service.pec.internalTrafficPolicy is set to local, filter and return the addresses
+
+	// If endpoint filtering is disabled, return all available addresses.
+	if !et.enableEndpointFiltering {
+		for k, v := range et.availableEndpoints.Addresses {
+			filtered[k] = v
+		}
+		return watcher.AddressSet{
+			Addresses:          filtered,
+			Labels:             et.availableEndpoints.Labels,
+			LocalTrafficPolicy: et.availableEndpoints.LocalTrafficPolicy,
```

Comment:
What does LocalTrafficPolicy do for us in this case? Why do we need to set it in this case? it seems like this sets us up to tell the consumer there's a local traffic policy when we actually have not honored it (below).

Reply:
The LocalTrafficPolicy field is only used by the `filterAddresses` function, later in a code path that we don't hit from here.  So the value of this field is unused and setting it is pointless.  I've removed setting the field here.